File: 프로그래머스/lv1/42840. 모의고사/모의고사.java
import java.util.*;


class Solution {
    public int[] solution(int[] answers) {

        int[] no1Ans = {1, 2, 3, 4, 5};
        int[] no2Ans = {2, 1, 2, 3, 2, 4, 2, 5};
        int[] no3Ans = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
        int[] score = new int[3];
        
        for (int i = 0; i < answers.length; i++) {
            if (no1Ans[i%no1Ans.length] == answers[i]) {
                    score[0]++;
            }
            if (no2Ans[i%no2Ans.length] == answers[i]) {
                    score[1]++;
            }
            if (no3Ans[i%no3Ans.length] == answers[i]) {
                    score[2]++;
            }
        }

        int maxScore = Math.max(score[0], Math.max(score[1], score[2]));
        ArrayList<Integer> list = new ArrayList<>();
        if(maxScore == score[0]) {
            list.add(1);
        }
        if(maxScore == score[1]) {
            list.add(2);
        }
        if(maxScore == score[2]) {
            list.add(3);
        }
        
        return list.stream().mapToInt(i->i.intValue()).toArray();
    }
}

File: 프로그래머스/lv1/42840. 모의고사/모의고사.py
def solution(answers):
    sooPoza1 = [1, 2, 3, 4, 5]
    sooPoza2 = [2,1,2,3,2,4,2,5]
    sooPoza3 = [3,3,1,1,2,2,4,4,5,5]
    d = {1:0, 2:0, 3:0}
    for i in range (len(answers)):
        if sooPoza1[i%5] == answers[i]:d[1] += 1
        if sooPoza2[i%8] == answers[i]:d[2] += 1
        if sooPoza3[i%10] == answers[i]:d[3] += 1

    grade =	[k for k,v in d.items() if max(d.values()) == v]

    return grade